src/mtype.d

```diff
@@ -9497,27 +9453,69 @@ public:
         return params;
     }
 
+    extern (D):
```

Comment:
I know a function taking D delegate cannot be `extern(C++)`. In this case it would be `_foreach`. But if so, why not you just add `extern (D)` to `_foreach` function?
More badly, you're wrongly change `isTPL` function to `extern (D)`. But in `mtype.h` it's still `extern (C++)`. It's a mismatch.